ios/VisionCameraPluginInatVision/VisionCameraPluginInatVision.m

```diff
@@ -6,157 +6,194 @@
+    
+    MLMultiArray *results = nil;
+    
+
+    if (geoModelPreds != nil) {
+        NSError *err = nil;
+        results = [self combineVisionScores:visionScores with:geoModelPreds error:&err];
+        results = [self normalizeMultiArray:results error:&err];
+    } else {
+        results = visionScores;
+    }
+
+    
+    // Setup taxonomy
+    VCPTaxonomy *taxonomy = [VisionCameraPluginInatVisionPlugin taxonomyWithTaxonomyFile:taxonomyPath];
+
+    NSMutableArray *topBranches = [NSMutableArray array];
+    NSArray *bestBranch = [taxonomy inflateTopBranchFromClassification:results];
     // add this to the end of the recent top branches array
     [topBranches addObject:bestBranch];
-  };
-
-  VNCoreMLRequest *objectRecognition = [[VNCoreMLRequest alloc] initWithModel:visionModel
-                                                            completionHandler:recognitionHandler];
-  objectRecognition.imageCropAndScaleOption = VNImageCropAndScaleOptionCenterCrop;
-  NSArray *requests = @[objectRecognition];
-
-  VNImageRequestHandler *handler = [[VNImageRequestHandler alloc] initWithCVPixelBuffer:pixelBuffer
-                                                                            orientation:orientation
-                                                                                options:@{}];
-  NSError *requestError = nil;
-  [handler performRequests:requests
-                      error:&requestError];
-  if (requestError) {
-      NSString *errString = [NSString stringWithFormat:@"got a request error: %@",
-                              requestError.localizedDescription];
-      NSLog(@"%@", errString);
-      return nil;
-  }
-
-  // convert the VCPPredictions in the bestRecentBranch into dicts
-  NSMutableArray *bestBranchAsDict = [NSMutableArray array];
-  for (VCPPrediction *prediction in topBranches.firstObject) {
-      [bestBranchAsDict addObject:[prediction asDict]];
-  }
-
-  // Create a new dictionary with the bestBranchAsDict under the key "predictions"
-  NSDictionary *response = [NSDictionary dictionary];
-  response = @{@"predictions": bestBranchAsDict};
-
-  // End timestamp
-  NSTimeInterval timeElapsed = [[NSDate date] timeIntervalSinceDate:startDate];
-  NSLog(@"inatVision took %f seconds", timeElapsed);
-
-  return response;
+    
+    // convert the VCPPredictions in the bestRecentBranch into dicts
+    NSMutableArray *bestBranchAsDict = [NSMutableArray array];
+    for (VCPPrediction *prediction in topBranches.firstObject) {
+        [bestBranchAsDict addObject:[prediction asDict]];
+    }
+    
+    // Create a new dictionary with the bestBranchAsDict under the key "predictions"
+    NSDictionary *response = [NSDictionary dictionary];
+    response = @{@"predictions": bestBranchAsDict};
+    
+    // End timestamp
+    NSTimeInterval timeElapsed = [[NSDate date] timeIntervalSinceDate:startDate];
+    NSLog(@"inatVision took %f seconds", timeElapsed);
```

Comment:
It would be lovely to return this telemetry to the caller. Perhaps in the response dictionary?

Reply:
good call. this is in 760d3ab09f3c993d795a5738f1190667520ebab4